booster/test/rpc-integration/test-vacuous/response-vacuous-but-rewritten.json

```diff
@@ -2,184 +2,136 @@
     "jsonrpc": "2.0",
     "id": 1,
     "result": {
-        "reason": "vacuous",
+        "reason": "stuck",
         "depth": 1,
```

Comment:
This might be a deal-breaker for the PR:
* without `--post-exec-simplify`, `kore-rpc-booster` does not recognise that the result is vacuous (the input state was already  vacuous, having `N ==Int 1 andBool N =/=Int 1` in the predicate). Also, the split between predicates and substitutions is not happening in the same way as in `kore-rpc` / fallbacks (see other tests).

Reply:
I think we are unlikely to end-up in this situation in the first place, because Booster checks consistency of ensured conditions and would report if it makes the pattern `#Bottom` during rewriting, i.e. executing the state
  ```
  Term:
      <generatedTop>(
          <k>(kseq("init", dotk())),
          <int>("0"),
          <generatedCounter>("0")
      )
  Conditions:
  ```

  results in a vacuous response after an attempt to apply the `init` rule:
  ```
  "result": {"reason":"vacuous","depth":0, ...
  ```

  and the logs would show:

  ```
  [booster][execute][term e7bdd85][rewrite 29b5f78][detail] TEST.init
  [booster][execute][term e7bdd85][rewrite 29b5f78][match][success] TEST.init
  [booster][execute][term e7bdd85][rewrite 29b5f78][match][success][substitution] Substitution: Rule#Var_DotVar0:SortGeneratedCounterCell{} -> <generatedCounter>("0") , Rule#VarX:SortInt{} -> "0" , Rule#Var_Gen4:SortIntCell{} -> <int>("0") , Rule#Var_DotVar1:SortK{} -> dotk()
  [booster][execute][term e7bdd85][rewrite 29b5f78][constraint][term d6e7b73][kore-term] _==Int_("0", "42")
  [booster][execute][term e7bdd85][rewrite 29b5f78][constraint][term d6e7b73][hook INT.eq][success][term 9eb81e8][kore-term] "false"
  [booster][execute][term e7bdd85][rewrite e7fa1c1][detail] TEST.JNIT-E
  [booster][execute][term e7bdd85][rewrite e7fa1c1][match][failure] Values differ:"jnit" "init"
  [booster][execute][term e7bdd85][rewrite 928fbc2][detail] TEST.JNIT-F
  [booster][execute][term e7bdd85][rewrite 928fbc2][match][failure] Values differ:"jnit" "init"
  [booster][execute][term e7bdd85][kore-term] <generatedTop>(<k>(kseq("init", dotk())), <int>("0"), <generatedCounter>("0"))
  [booster][execute][term e7bdd85] Simplified to bottom after 0 steps.
  ```

We are fine in `"execute"` requests, because Booster would never produce `state-vacuous-but-rewritten.execute` anymore. It would have before, as we did not have the SMT solver to prune the contradictory ensured conditions.

I have a related comment about the `"simplify"` requests, but I'll post it in a separate comment.